File: src/main/java/com/uwetrottmann/trakt5/services/People.java
/*
 * Copyright 2024 Uwe Trottmann
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.uwetrottmann.trakt5.services;

import com.uwetrottmann.trakt5.entities.Credits;
import com.uwetrottmann.trakt5.entities.Person;
import com.uwetrottmann.trakt5.enums.Extended;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface People {

    /**
     * Returns a single person's details.
     *
     * @param personId trakt ID, trakt slug, or IMDB ID Example: bryan-cranston.
     */
    @GET("people/{id}")
    Call<Person> summary(
            @Path("id") String personId,
            @Query("extended") Extended extended
    );

    @GET("people/{id}/movies")
    Call<Credits> movieCredits(
            @Path("id") String personId
    );

    @GET("people/{id}/shows")
    Call<Credits> showCredits(
            @Path("id") String personId
    );

}

File: src/main/kotlin/com/redissi/trakt/services/People.kt
package com.redissi.trakt.services

import com.redissi.trakt.entities.Credits
import com.redissi.trakt.entities.Person
import com.redissi.trakt.enums.Extended
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface People {
    /**
     * Returns a single person's details.
     *
     * @param personId trakt ID, trakt slug, or IMDB ID Example: bryan-cranston.
     */
    @GET("people/{id}")
    suspend fun summary(
        @Path("id") personId: String?,
        @Query("extended") extended: Extended?
    ): Person?

    @GET("people/{id}/movies")
    suspend fun movieCredits(
        @Path("id") personId: String?
    ): Credits?

    @GET("people/{id}/shows")
    suspend fun showCredits(
        @Path("id") personId: String?
    ): Credits?
}